lib/restrictions.js

```diff
@@ -24,6 +24,17 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
     collection: 'resource-restriction-restriction',
     fields: {'restriction.zone': 1},
     options: {unique: false, background: false}
+  },
+  {
+    // ensure that each restriction is unique
+    collection: 'resource-restriction-restriction',
+    fields: {
+      'restriction.zone': 1,
+      'restriction.resource': 1,
+      'restiction.methodOptions.limit': 1,
+      'restiction.methodOptions.duration': 1
+    },
+    options: {unique: false, background: false}
```

Comment:
This is not `unique` (it is set to `false`). Also, this can't be done with uniqueness on `restriction.id` as well without disabling sharding. We should consider alternative fixes to whatever the problem is.

Reply:
@dlongley what we are trying to prevent is someone being able to add the same restriction to the system over and over again, which they are now able to do if they use a different id each time.